components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/EmailTemplateManagerImpl.java

```diff
@@ -723,6 +689,98 @@ public boolean isEmailTemplateTypeExists(String templateTypeDisplayName, String
+
+        String templateType = I18nEmailUtil.getNormalizedName(templateTypeName);
+        String path = EMAIL_TEMPLATE_PATH + PATH_SEPARATOR + templateType + getApplicationPath(applicationUuid);
+        try {
+            resourceMgtService.deleteIdentityResource(path, tenantDomain, localeCode);
+        } catch (IdentityRuntimeException ex) {
+            String msg = String.format("Error deleting %s:%s template from %s tenant registry.", templateTypeName,
+                    localeCode, tenantDomain);
+            handleServerException(msg, ex);
+        }
+    }
+
+    @Override
+    public EmailTemplate getEmailTemplate(String templateType, String locale, String tenantDomain,
+                                          String applicationUuid) throws I18nEmailMgtException {
+
+        try {
+            NotificationTemplate notificationTemplate = getNotificationTemplate(
+                    NotificationChannels.EMAIL_CHANNEL.getChannelType(), templateType, locale,
+                    tenantDomain, applicationUuid);
+            return buildEmailTemplate(notificationTemplate);
+        } catch (NotificationTemplateManagerException exception) {
+            String errorCode = exception.getErrorCode();
+            String errorMsg = exception.getMessage();
+            Throwable throwable = exception.getCause();
+
+            // Match NotificationTemplateManagerExceptions with the existing I18nEmailMgtException error types.
+            if (StringUtils.isNotEmpty(exception.getErrorCode())) {
+                if (IdentityMgtConstants.ErrorMessages.ERROR_CODE_INVALID_NOTIFICATION_TEMPLATE.getCode()
+                        .equals(errorCode) || IdentityMgtConstants.ErrorMessages.ERROR_CODE_NO_CONTENT_IN_TEMPLATE
+                        .getCode().equals(errorCode) ||
+                        I18nMgtConstants.ErrorMessages.ERROR_CODE_INVALID_CHARACTERS_IN_TEMPLATE_NAME.getCode()
+                                .equals(errorCode) ||
+                        I18nMgtConstants.ErrorMessages.ERROR_CODE_INVALID_CHARACTERS_IN_LOCALE
+                                .getCode().equals(errorCode)) {
+                    throw new I18nEmailMgtClientException(errorMsg, throwable);
+                } else if (IdentityMgtConstants.ErrorMessages.ERROR_CODE_INVALID_EMAIL_TEMPLATE_CONTENT.getCode()
+                        .equals(errorCode)) {
+                    throw new I18nMgtEmailConfigException(errorMsg, throwable);
+                } else if (IdentityMgtConstants.ErrorMessages.ERROR_CODE_NO_TEMPLATE_FOUND.getCode()
+                        .equals(errorCode)) {
+                    throw new I18nEmailMgtInternalException(I18nMgtConstants.ErrorCodes.EMAIL_TEMPLATE_TYPE_NODE_FOUND,
+                            errorMsg, throwable);
+                }
+            }
+            throw new I18nEmailMgtServerException(exception.getMessage(), exception.getCause());
+        }
+    }
+
+    private String getApplicationPath(String applicationUuid) {
+
```

Comment:
Add a comment to explain the scenario

Reply:
Addressed with https://github.com/wso2-extensions/identity-event-handler-notification/pull/223/commits/43afd46f6a19a8fc329ccee7166d75003c866306